src/client_capabilities.ts

```diff
@@ -0,0 +1,111 @@
+import { isVersion } from './is_version.ts';
+
+export function prepareGetClientCapabilities(ua: string = '') {
+  let version: isVersion;
+  try {
+    /**
+     * Make sure at least PublicKeyCredential is defined before trying to polyfill anything on it
+     */
+    version = new isVersion(ua);
+  } catch {
+    console.info('expected exception for the test.');
+  }
+
+  return async function () {
+    let conditionalCreate: boolean | undefined = false;
+    let conditionalGet: boolean | undefined = false;
+    // `hybridTransport` is unsupported since it requires Bluetooth
+    // detection.
+    let hybridTransport: boolean | undefined = undefined;
+    // `passkeyPlatformAuthenticator` is unsupported either since
+    // `hybridTransport` is unavailable.
+    let passkeyPlatformAuthenticator: boolean | undefined = undefined;
+    let relatedOrigins: boolean | undefined = false;
+    let signalAllAcceptedCredentials: boolean | undefined = false;
+    let signalCurrentUserDetails: boolean | undefined = false;
+    let signalUnknownCredential: boolean | undefined = false;
+    let userVerifyingPlatformAuthenticator: boolean | undefined = false;
+
+    // If this is above macOS Safari 17.4 and below Safari 18.2, or above iOS 17.4 and below iOS 18.2, there's a spec glitch.
+    const capabilities = version.safari174To182 || version.iOS174To182
+      // @ts-ignore: We're polyfilling this, so ignore whether TS knows about this or not
+      ? await PublicKeyCredential.getClientCapabilities()
+      : undefined;
```

Comment:
It's unclear to me what Safari/iOS glitch is being worked around here. Is this intending to **skip** calling `PublicKeyCredential.getClientCapabilities()` in Safari iOS 17.4 up until 18.2? Because the way this ternary is set currently then the method will actually be called if we're looking at iOS 17.4 up until 18.2.

Reply:
Updated the line 29 so it better describes what it does.

Reply:
Ah, but reading the code there's nothing here actually mapping `capabilities.conditionalMediation` to `capabilities.conditionalGet` - it's the check on L43-L46 that's why `conditionalGet` is coming back `true` in the test above...

Reply:
🙄 Let's fix it in the next iteration